src/__tests__/execution.test.ts

```diff
@@ -76,7 +76,7 @@ describe("Execute: Handles basic execution tasks", () => {
       [],
       []
     );
-    await Promise.resolve(); // For the promise to resolve
+    await new Promise(r => setImmediate(r)); // For the promise to resolve
```

Comment:
This is because the Promises execution is now back to how it was in Node 8 and is now an optimization in Node 12 - 1 microtask instead of 3. 

So, we wait longer to support both node 10 and 12.